dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -24,6 +26,15 @@ def all
       # Note that @user.email is populated by User.from_omniauth even for students
       redirect_to "/users/sign_in?providerNotLinked=#{@user.provider}&email=#{@user.email}"
     else
+      # This is a new registration
+      # Because some oauth tokens are quite large, we strip the token from the session
+      # variables and pass them through via the cache instead - they are pulled out again
+      # from User::new_with_session
+      oauth_token = @user.attributes['properties']['oauth_token']
+      @user.attributes['properties']['oauth_token'] = nil
+      cache = CDO.shared_cache
+      oauth_cache_key = "oauth_token_#{@user.email}"
```

Comment:
Should this be a method? Here, "this" refers to both `oauth_cache_key` and the `cache.write` from the key and token.